homeassistant/components/lamarzocco/__init__.py

```diff
@@ -99,18 +106,27 @@ def bluetooth_configured() -> bool:
                 address_or_ble_device=entry.data[CONF_MAC],
             )
 
-    coordinator = LaMarzoccoUpdateCoordinator(
-        hass=hass,
-        entry=entry,
-        local_client=local_client,
+    device = LaMarzoccoMachine(
+        model=entry.data[CONF_MODEL],
+        serial_number=entry.unique_id,
+        name=entry.data[CONF_NAME],
         cloud_client=cloud_client,
+        local_client=local_client,
         bluetooth_client=bluetooth_client,
     )
 
-    await coordinator.async_config_entry_first_refresh()
-    entry.runtime_data = coordinator
+    coordinators = (
+        LaMarzoccoConfigUpdateCoordinator(hass, entry, device, local_client),
+        LaMarzoccoFirmwareUpdateCoordinator(hass, entry, device),
+        LaMarzoccoStatisticsUpdateCoordinator(hass, entry, device),
+    )
+
+    for coordinator in coordinators:
+        await coordinator.async_config_entry_first_refresh()
+
+    entry.runtime_data = LaMarzoccoRuntimeData(*coordinators)
```

Comment:
Wouldn't it make more sense to instantiate / add the coordinator directly into `LaMarzoccoRuntimeData` instead of making an tuple first?

Reply:
then I'd have to do something in the dataclass in a `post_init` or similar so make it so I can loop over them. So I thought might as well built a list first

Reply:
alright, coming up

Reply:
done.